lib/file-helper.js

```diff
@@ -16,13 +17,22 @@ const getListing = async (directoryPath) => {
 
   // Gather listing entries
   const entries = await Promise.all(basenames.map(async basename => {
-    const entryPath = join(directoryPath, basename)
+    const entryPath = normalize(join(directoryPath, basename))
 
-    return [basename, {
+    // File or directory entry
+    /** @type {DirectoryListingEntry} */
+    const entry = {
       basename,
-      path: normalize(entryPath),
+      path: entryPath,
       stats: await stat(entryPath)
-    }]
+    }
+
+    // Follow child directory listing
+    if (entry.stats.isDirectory()) {
+      entry.entries = await getListing(entryPath)
```

Comment:
This tiny change means we can remove [`recursive-readdir`](https://www.npmjs.com/package/recursive-readdir) and our `getFilesByDirectory()` helper function